enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -54,20 +54,22 @@ var (
-	enableCompressionFlag     = flag.Bool("cache.pebble.enable_zstd_compression", false, "If set, zstd compressed files can be saved to the cache. Otherwise only decompressed bytes should be stored.")
+	migrateFromDiskDir              = flag.String("cache.pebble.migrate_from_disk_dir", "", "If set, attempt to migrate this disk dir to a new pebble cache")
+	forceAllowMigration             = flag.Bool("cache.pebble.force_allow_migration", false, "If set, allow migrating into an existing pebble cache")
+	clearCacheBeforeMigration       = flag.Bool("cache.pebble.clear_cache_before_migration", false, "If set, clear any existing cache content before migrating")
+	scanForOrphanedFiles            = flag.Bool("cache.pebble.scan_for_orphaned_files", false, "If true, scan for orphaned files")
+	orphanDeleteDryRun              = flag.Bool("cache.pebble.orphan_delete_dry_run", true, "If set, log orphaned files instead of deleting them")
+	dirDeletionDelay                = flag.Duration("cache.pebble.dir_deletion_delay", time.Hour, "How old directories must be before being eligible for deletion when empty")
+	atimeUpdateThresholdFlag        = flag.Duration("cache.pebble.atime_update_threshold", DefaultAtimeUpdateThreshold, "Don't update atime if it was updated more recently than this")
+	atimeWriteBatchSizeFlag         = flag.Int("cache.pebble.atime_write_batch_size", DefaultAtimeWriteBatchSize, "Buffer this many writes before writing atime data")
+	atimeBufferSizeFlag             = flag.Int("cache.pebble.atime_buffer_size", DefaultAtimeBufferSize, "Buffer up to this many atime updates in a channel before dropping atime updates")
+	minEvictionAgeFlag              = flag.Duration("cache.pebble.min_eviction_age", DefaultMinEvictionAge, "Don't evict anything unless it's been idle for at least this long")
+	forceCompaction                 = flag.Bool("cache.pebble.force_compaction", false, "If set, compact the DB when it's created")
+	forceCalculateMetadata          = flag.Bool("cache.pebble.force_calculate_metadata", false, "If set, partition size and counts will be calculated even if cached information is available.")
+	isolateByGroupIDsFlag           = flag.Bool("cache.pebble.isolate_by_group_ids", false, "If set, filepaths and filekeys for AC records will include groupIDs")
+	enableCompressionFlag           = flag.Bool("cache.pebble.enable_zstd_compression", false, "If set, zstd compressed files can be saved to the cache. Otherwise only decompressed bytes should be stored.")
+	compressWritesFlag              = flag.Bool("cache.pebble.compress_writes", false, "If set, data should be compressed before written to disk")
+	minCompressionFileSizeBytesFlag = flag.Int64("cache.pebble.min_compression_file_size_bytes", DefaultMaxInlineFileSizeBytes, "If compression is on, files larger than this will be compressed before written to disk.")
```

Comment:
So by setting this to DefaultMaxInlineFileSizeBytes, we're saying we'll compress everything not inlined, i.e. not written to pebble. Which seems OK because by default pebble is already using snappy compression: https://github.com/cockroachdb/pebble/blob/4a63cdb3a71eb3a58d1a8a9849b6314e7799c6a1/options.go#L402

I think we should add a flag to use zstd for pebble stuff too before migrating. It's about as fast as snappy and has a slightly better compression ratio

Reply:
Okay I'll add the inlining zstd in a separate PR so this one isn't blocked while I test that

Reply:
@tylerwilliams looks like zstd is not as performant in pebble. Probably not worth it for minor compression ratio improvements: https://github.com/cockroachdb/pebble/issues/1582

"Writer throughput with zstd is still ~70% slower than with snappy."